eman2/protocols/protocol_tomo_boxing.py

```diff
@@ -0,0 +1,189 @@
+# *
+# * This program is distributed in the hope that it will be useful,
+# * but WITHOUT ANY WARRANTY; without even the implied warranty of
+# * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# * GNU General Public License for more details.
+# *
+# * You should have received a copy of the GNU General Public License
+# * along with this program; if not, write to the Free Software
+# * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+# * 02111-1307  USA
+# *
+# *  All comments concerning this program package may be sent to the
+# *  e-mail address 'scipion@cnb.csic.es'
+# *
+# **************************************************************************
+
+from pyworkflow.utils.properties import Message
+from pyworkflow.gui.dialog import askYesNo
+from pyworkflow.protocol.params import BooleanParam
+from pyworkflow import utils as pwutils
+
+import eman2
+from eman2.convert import loadJson
+
+from tomo.protocols.protocol_base import ProtTomoBoxing
+from tomo.objects import Coordinate3D
+
+
+class EmanProtTomoBoxing(ProtTomoBoxing):
+    """ Manual picker for Tomo. Uses EMAN2 e2spt_boxer.py.
+    """
+    _label = 'tomo boxer'
+
+    def __init__(self, **kwargs):
+        ProtTomoBoxing.__init__(self, **kwargs)
+
+    @classmethod
+    def isDisabled(cls):
+        return not eman2.Plugin.isNewVersion()
+
+    # --------------------------- DEFINE param functions ----------------------
+    def _defineParams(self, form):
+        ProtTomoBoxing._defineParams(self, form)
+
+        form.addParam('inMemory', BooleanParam, default=False,
+                      label='Read in Memory',
+                      help='This will read the entire tomogram into memory.'
+                           'Much faster, but you must have enough ram.')
+
+    # --------------------------- INSERT steps functions ----------------------
+    def _insertAllSteps(self):
+        self.inputTomo = self.inputTomogram.get()
+        self._params = {'inputTomogram': self.inputTomo.getFileName()}
+        # Launch Boxing GUI
+        self._insertFunctionStep('launchBoxingGUIStep', self._params, interactive=True)
+
+    def _readCoordinates3D(self, box, tomo, coord3DSet):
+        x, y, z = box[:3]
+        coord = Coordinate3D()
+        coord.setPosition(x, y, z)
+        coord.setVolume(tomo)
+        coord3DSet.append(coord)
+
+    def _readSetOfCoordinates3D(self, jsonBoxDict, tomo, coord3DSetDict):
+        if jsonBoxDict.has_key("boxes_3d"):
+            boxes = jsonBoxDict["boxes_3d"]
+
+            for box in boxes:
+                classKey = box[5]
+                coord3DSet = coord3DSetDict[classKey]
+                coord3DSet.enableAppend()
+
+                self._readCoordinates3D(box, tomo, coord3DSet)
+
+    def _createOutput(self, outputDir):
+        jsonFnbase = pwutils.join(outputDir, 'info',
+                                  'extra-%s_info.json' % pwutils.removeBaseExt(self.inputTomo.getFileName()))
```

Comment:
Check line length

Reply:
This was not highlighted for me as a PEP8 error...weird... anyway I have just split the line